gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFSInputStream.java

```diff
@@ -110,12 +129,18 @@ public synchronized int read() throws IOException {
    */
   @Override
   public synchronized int read(byte[] buf, int offset, int length) throws IOException {
+    Stopwatch stopwatch = Stopwatch.createStarted();
+    Map<String, Object> apiTraces = new HashMap<>();
```

Comment:
It would be great if we can avoid creating this object in case tracing in not enabled. We are paying the price in all code paths.

Reply:
I think it is ok to have another method to save on this. Note that tracing will be disabled on most of the cases.